malcolm/modules/ADCore/parts/detectordriverpart.py

```diff
@@ -144,13 +145,19 @@ def wait_for_detector(
         child.arrayCounterReadback.subscribe_value(
             self.update_completed_steps, registrar
         )
-        # If no new frames produced in event_timeout seconds, consider scan dead
-        context.wait_all_futures(self.start_future, event_timeout=event_timeout)
+
+        if not self.is_hardware_triggered:
+            # If soft triggered then wait for the detector to stop acquiring before
+            # checking if all frames were received
+            context.wait_all_futures(self.start_future, event_timeout=event_timeout)
```

Comment:
I reckon we should never bother storing and checking this start_future, we can't check it in the hardware triggered case so there's not much to be gained.

Reply:
Removed in 1d1488e